lib/stanford_core/tree.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module StanfordCore
+  class Tree
+    VERB_PHRASE = 'VP'
+    NOUN_PHRASE = 'NP'
+    VERB = 'VB'
+    VERB_PAST_TENSE = 'VBD'
+    VERB_PAST_PARTICIPLES = 'VBN'
+    VERB_GERUND = 'VBG'
+    PROPER_NOUN = 'NNP'
+    PREPOSITION_PHRASE = 'PP'
+
+    def initialize(tree)
+      @tree = tree
+    end
+
+    def verb_phrase_with_noun?
+      verb_phrase? && second_child_is_noun_phrase?
+    end
+
+    def verb_phrase?
+      label_value == VERB_PHRASE
+    end
+
+    def preposition_phrase?
+      label_value == PREPOSITION_PHRASE
+    end
+
+    def leaf?
+      @tree.is_leaf
+    end
+
+    def verb?
+      [VERB, VERB_PAST_TENSE, VERB_PAST_PARTICIPLES, VERB_GERUND]
+        .include?(label_value)
+    end
+
+    def proper_noun?
+      label_value == PROPER_NOUN
+    end
+
+    def noun_phrase?
+      label_value == NOUN_PHRASE
+    end
+
+    def to_s
+      return @tree.to_s if leaf?
+
+      @to_s ||= begin
+        list = []
+        each_leave do |leaf|
+          list.push(leaf.to_s)
+        end
+        list.join(' ').gsub(' .', '.').gsub(' ,', ',')
+      end
+    end
+
+    def lemma_string
+      Text.new(to_s).lemmas.join(' ')
+    end
+
+    def generalized_string
+      list = []
+      each_child do |subtree|
+        if subtree.proper_noun?
+          list.push('someone')
+        elsif subtree.leaf?
+          list.push(subtree.to_s)
+        else
+          list.push(subtree.generalized_string)
+        end
+      end
+      list.chunk(&:itself).map(&:first).join(' ')
+    end
```

Comment:
These two methods have to be refactored.
They are doing what they should, and are being tested, as proven by the code coverage report.

However, they will probably be changed later in this project. That's why I am not adding specific tests for them.